### PR TITLE
Updated hero page

### PR DESCRIPTION
**Pull request by ChristopherChudzicki**
### What are the relevant tickets?
- https://github.com/mitodl/hq/issues/4375

### Description (What does it do?)
Updates the hero section

### Screenshots (if appropriate):

Screenshot 2024-05-29 at 3 39 16 PM

Screenshot 2024-05-29 at 3 39 23 PM

https://github.com/mitodl/mit-open/assets/9010790/b1015d51-7c72-4da9-b9d0-f58757fb04eb

### How can this be tested?
1. View the homepage. Its hero section should look like [designs](https://www.figma.com/design/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?node-id=3852-52752&m=dev)
2. Click the links. 
    - "trending" links should go to appropriate search page
    - browse by topics => topics parge
    - explore all => search page

### CODE REVIEW

**Comment by Ferdi** [comment]
@ChristopherChudzicki you love those "AI courses"  :-)  

That should be "Browse by Topics" instead , and link to the Topics page.

File: frontends/ol-components/src/components/Input/Input.tsx [hunk 1]
```diff
@@ -52,10 +52,10 @@ const Input = styled(InputBase)(({
       paddingRight: "12px",
       borderRadius: "4px",
       "&.MuiInputBase-adornedStart": {
-        marginLeft: `-${buttonPadding.medium}px`,
+        paddingLeft: `${12 - buttonPadding.medium}px`,
```
Comment (ChristopherChudzicki) [inline frontends/ol-components/src/components/Input/Input.tsx:55]: The idea here was:
- the AdornmentButtons are a bit bigger than designs specify... they include some extra padding to make interacting easier.
- So when there are adornment buttons, add a bit less padding. Originally I was using negative margin, but I had it pon the wrong element.

File: frontends/mit-open/src/pages/HomePage/HeroSearch.tsx [hunk 2]
```diff
@@ -80,41 +177,67 @@ const HeroSearch: React.FC = () => {
...
+            size={isMobile ? "medium" : "hero"}
+            fullWidth
+            value={searchText}
+            onChange={onSearchChange}
+            onClear={onSearchClear}
+            onSubmit={onSearchSubmit}
+          />
+          <LinksContainer>
+            <TrenderingContainer>
+              <Typography typography={{ xs: "subtitle4", md: "subtitle3" }}>
```
Comment (gumaerc) [inline frontends/mit-open/src/pages/HomePage/HeroSearch.tsx:204]: We may want to add some margin to the right of the "Trending" text, as sometimes it runs up right against the edge of the search chip next to it.

image

File: frontends/mit-open/src/pages/HomePage/HeroSearch.tsx [hunk 3]
```diff
@@ -19,49 +18,147 @@ type SearchChip = {
 
 const SEARCH_CHIPS: SearchChip[] = [
   {
-    label: "AI Courses",
-    href: "/search?topic=Artificial+Intelligence&resource_type=course",
+    label: "New",
+    href: "/search?sortby=new",
```
Comment (gumaerc) [inline frontends/mit-open/src/pages/HomePage/HeroSearch.tsx:22]: I know this isn't mentioned in the issue, but do we want to maybe filter this on "course" as well? For example, the "New" item in the nav menu points to `?resource_type=course&sortby=new` to make sure only courses are returned. Did we want to include all learning resources here or just courses?

File: frontends/mit-open/src/pages/HomePage/HeroSearch.tsx [hunk 4]
```diff
@@ -19,49 +18,147 @@ type SearchChip = {
...
-  display: flex;
-  flex-direction: column;
-  align-items: flex-start;
-  justify-content: center;
-  ${({ theme }) => theme.breakpoints.down("sm")} {
-    min-height: 272px;
-  }
-`
+const TitleAndControls = styled.div(({ theme }) => ({
+  // flex: 1;
```
Comment (gumaerc) [inline frontends/mit-open/src/pages/HomePage/HeroSearch.tsx:136]: Commented out styles here should be removed

**Review by gumaerc — CHANGES_REQUESTED** [review]
This looks good, but I have some questions and nit picks:

File: frontends/mit-open/src/pages/HomePage/HeroSearch.tsx (hunk 3 above)
Comment (gumaerc) [reply frontends/mit-open/src/pages/HomePage/HeroSearch.tsx:22]: (or, maybe we should change the nav menu and remove the course filter, not sure... either way it should probably be consistent)

Comment (ChristopherChudzicki) [reply frontends/mit-open/src/pages/HomePage/HeroSearch.tsx:22]: We should change the nav drawer. Confirmed with Ferdi, and added to this PR.

https://github.com/mitodl/hq/issues/4179#issuecomment-2103190033

**Review by gumaerc — APPROVED** [review]
👍

**Pull request merged**